odl/discr/partition.py

```diff
@@ -0,0 +1,561 @@
+
+        >>> part = RectPartition(grid, begin=[0, -1], end=[1, 2])
+        >>> part.min()
+        array([ 0., -1.])
+
+        See also
+        --------
+        bbox
+        """
+        return self.bbox.min()
+
+    def max(self):
+        """Return the 'minimum corner' of ``self.interv_prod``.
+
+        Examples
+        --------
+        >>> grid = TensorGrid([0, 1], [-1, 0, 2])
+        >>> part = RectPartition(grid)  # Implicit begin and end
+        >>> part.max()
+        array([ 1.5,  3. ])
+
+        >>> part = RectPartition(grid, begin=[0, -1], end=[1, 2])
+        >>> part.max()
+        array([ 1.,  2.])
+
+        See also
+        --------
+        bbox
+        """
+        return self.bbox.max()
+
+    def extent(self):
+        """Return a vector containing the total extent (max - min)."""
+        return self.max() - self.min()
+
+    def nodes_on_boundary(self):
+        """Return a list indicating if nodes lie on the boundary.
+
+        Returns
+        -------
+        on_bdry : `tuple` of 2-tuple of `bool`
+            Each 2-tuple contains the information whether the leftmost
+            (first entry) and rightmost (second entry) grid nodes lie
+            on the boundary of the bounding box in the corresponding
+            dimension.
+
+        Examples
+        --------
+        >>> grid = TensorGrid([0, 1], [-1, 0, 2])
+        >>> part = RectPartition(grid, begin=[0, -2], end=[2, 2])
+        >>> part.nodes_on_boundary()
+        ((True, False), (False, True))
+        """
+        bdry_list = []
+        for ax, (minpt, maxpt, bmin, bmax) in enumerate(zip(
+                self.grid.min_pt, self.grid.max_pt,
+                self.bbox.begin, self.bbox.end)):
+            bdry_list.append((minpt == bmin, maxpt == bmax))
+
+        return tuple(bdry_list)
+
+    def sampling_points(self):
+        """Return the sampling grid points."""
+        return self.grid.points()
+
+    def sampling_meshgrid(self):
+        """Return the sparse meshgrid of sampling points."""
+        return self.grid.meshgrid()
+
+    def cell_boundaries(self):
+        """Return the cell boundaries as coordinate vectors.
+
+        Examples
+        --------
+        >>> grid = TensorGrid([0, 1], [-1, 0, 2])
+        >>> part = RectPartition(grid)  # Implicit begin and end
+        >>> part.cell_boundaries()
+        (array([-0.5,  0.5,  1.5]), array([-1.5, -0.5,  1. ,  3. ]))
+
+        >>> part = RectPartition(grid, begin=[0, -1], end=[1, 2])
+        >>> part.cell_boundaries()
+        (array([ 0. ,  0.5,  1. ]), array([-1. , -0.5,  1. ,  2. ]))
+        """
+        bdry_vecs = []
+        for ax, vec in enumerate(self.grid.coord_vectors):
+            bdry = np.empty(len(vec) + 1)
+            bdry[1:-1] = (vec[1:] + vec[:-1]) / 2.0
+            bdry[0] = self.min()[ax]
+            bdry[-1] = self.max()[ax]
+            bdry_vecs.append(bdry)
+
+        return tuple(bdry_vecs)
+
+    def cell_sizes(self):
```

Comment:
As discussed above, sizes is quite confusing here, perhaps something indicating this is the side lengths.